woocommerce/woocommerce
# k6: Update existing tests for COT

### All Submissions:






### Changes proposed in this Pull Request:

This PR makes updates to the existing k6 tests
- Fixes all linting errors for `tests/performance`
- Updates some merchant requests to match latest requests in WC Core (Closes 349-gh-woocommerce/woocommerce-quality **k6: Check requests against 6.9**)
- Updates Merchant Order tests and config to be able to use COT URLs by using new `cot_status` config variable (Closes https://github.com/woocommerce/woocommerce/issues/34395)
- Removed existing simple shopper and merchant requests tests (used for quickly trying all the requests) and replaced them with a consolidated `simple-all-requests test`
- Removed custom metrics and replaced them with tags and thresholds (Closes 300-gh-woocommerce/woocommerce-quality **k6: Reduce Custom Metrics**)


More detail on replacing custom metrics with tags and thresholds.

With this PR: Tags added for request names, thresholds added to appear in summary report.
![Screenshot 2022-09-06 at 00 43 37](https://user-images.githubusercontent.com/24649833/188521305-fc345735-4c45-404c-a5e4-e00384c2c5c2.png)

Before this PR: Custom metrics used to add requests to summary report, see 300-gh-woocommerce/woocommerce-quality for why this is not advised.
![Screenshot 2022-09-06 at 00 44 24](https://user-images.githubusercontent.com/24649833/188521376-fcf3cffc-0929-45f9-a0e6-9192cfa6774f.png)


### How to test the changes in this Pull Request:

1. Setup test environment
[See instructions here for more details
](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/tests/performance#test-environment)
- use wp-env to setup a local environment with `performance/bin/init-sample-products.sh` that will set up a shop with sample products imported and the shop settings (payment method, permalinks, address etc) needed for the tests already set.
- OR use an existing test site and update `performance/config.js` to use that site


2. Install k6 and run test
run `brew install k6`
[For other platforms please see the k6 installation guide.](https://k6.io/docs/getting-started/installation/)
change directory to `plugins/woocommerce/tests/performance`
Run all requests with 
`k6 run tests/simple-all-requests.js`

### Other information:




### FOR PR REVIEWER ONLY:
